### DISCUSSION TITLE
Tail sampling - how to keep the root span even though not sampled

### DISCUSSION THREAD

**Discussion by kes2464** [open-telemetry/opentelemetry-collector-contrib#24557] [Q&A]
I am currently using Elastic and its sampling.
I would like to replace it with OpenTelemetry collector's tail sampling processor.

With tail sampling processor, when the spans are not sampled, it drops all spans with the same Trace ID. 
However Elastic sampling drops all children spans but keeps the root spans (transactions).

Is there any way to achieve the same behaviour as Elastic's sampling? 

I have tried using ottl_condition to filter the root span, but it ends up sampling all spans within the same context.

**Comment by kes2464** [comment]
Anyone has any idea?

**Comment by Vinaum8** [comment]
Hello everyone,

Is it a good practice to use groupbytrace in the processors of an open telemetry collector tracing?

`processors: [groupbytrace, tail_sampling]`

I am using tail sampling to exclude tracings that are shorter than <200ms, but some spans are coming with the value:

_<root span not yet received>_

Has anyone solved this problem?

**Comment by jayadeep-keka** [comment]
I am also facing this issue
